### DISCUSSION TITLE
Idea: Approve-only mode (no passwords)

### DISCUSSION THREAD

**Discussion by My1** [rustdesk/rustdesk#1869] [General]
so when starting this application behaves simiar to teamviewer with a randomly generated password.

while not only a pain to tell over the phone it comes with the bonus of not regenerating after the connection closes which means someone can just reconnect after being kicked out.

on anydesk to compare there is no password by default, you either have to approve locally, or you can make yourself unattended access using a password (and if wanted, a second factor).

This means that each connection that isnt going over unattended has to be approved by the user clicking on the button.

**Comment by lukasrajzl** [comment]
I would like this feuture in RustDesk too...

**Comment by pedjas** [comment]
Approve only mode should be default behavior. 

One should not be able to connect to the client without it being specifically allowed. Other than approve only mode there should be option to specifically turn on unattended mode.

Ideally it should have all three ways to connect:

- Approve only mode - user has to specificaly approve each connection to his computer
- Unattended mode - remote user can access if he enters password
- Changing password mode - password changes for each new connection

Reason is that many people are used to other remote access platforms that do it differently. People often are not tech oriented and it is hard for them to learn new ways.

For example, users of TeamViewer are used to share ID and password for remote access. Password changes on each connection so that is safe. When they switch to RustDesk and do the same, they actually provide unlimited access to other side, as sharing password means allowing uncontrolled remote access. People may be unaware that sharing password is actually unsafe.

Approve only mode is clean, simple, easy to understand and safest approach.

Unattended mode is needed for those that need to access their own computer from remote, and there is no one that could or should manually approve access each time. But that implies user is advanced and knows what he is doing.

**Reply by My1** [reply]
I need to double check but i think TeamViewer didn't change its password per connection but rather on each restart of teamviewer. However my last experience with teamviewer is quite a while ago.

Random passwords are in my opinion usually kinda dumb as they are a pain to transmit over the phone and an accept button is a lot easier.

**Reply by lukasrajzl** [reply]
Accept button is allready in  RustDesk, the only thing left is disable password or use random password that change after every connection. For me both variant (change password after connection or disable password login) are sufficient.

**Reply by pedjas** [reply]
> Random passwords are in my opinion usually kinda dumb as they are a pain to transmit over the phone and an accept button is a lot easier.

I agree, but that is how some people are used to do due to experience with other platforms.

However, with Approve only mode and hiding password for that mode makes changing password not needed option, as password would then be used only for unattended mode, which one who turns on knows how to setup and use. Setting password should then be hidden somewhere deeper in configuration, not at the main screen.

How it is now is very unsafe. Uneducated client may conclude he has to give password to the other side, not realizing that it means other side would then be able to access his computer whenever he wants without he even knowing that.

Safety must come first. RustDesk is excellent and it would be bad that it gets reputation that users can be easily tricked to give unattended access to their computers.

**Comment by 21pages** [comment]
current:
* password:
1. The random password will be regenerated each time the connection ends.
2. Users can set a permanent password.
3. Users can choose to use only temporary passwords or permanent passwords, or both
![1668499783872](https://user-images.githubusercontent.com/14891774/201864551-a2d334ca-56a7-45e7-945c-577e82b34c66.png)

* mode:
1. Click approve mode and password mode are both used at the same time
2. If you choose to use only a permanent password and set a empty permanent password, you will only be able to connect by clicking, which is a bit similar to Approve only mode

What about this? Use three modes:
1.  Approve only mode.
2.  Password only mode, both random password and pernament password are used default, you can set to disable random password, and it will be like unattended mode. 
3.  Both in use, this is default.

**Reply by rustdesk** [reply]
https://github.com/rustdesk/rustdesk/releases/tag/nightly

**Reply by ackid32** [reply]
Nice, but the single exe (I like this concept!) is missing. Any way to get all in a single executable?

**Reply by pedjas** [reply]
> Nice, but the single exe (I like this concept!) is missing. Any way to get all in a single executable?

This new version is actually single executable. Just rename rustdesk-1.2.0-x86_64-pc-windows-msvc.exe to insert host and key.

**Comment by 21pages** [comment]
#2247
